packages/react-component-library/src/components/Button/LoadingButtons.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useMemo, useState } from 'react'
+
+import { Button } from './Button'
+import { BUTTON_VARIANT } from './constants'
+
+import { COMPONENT_SIZE, ComponentSizeType } from '../Forms'
+
+export const LoadingButtons = ({
+  size = COMPONENT_SIZE.FORMS,
+}: {
+  size: ComponentSizeType
+}) => {
+  const allVariants = useMemo(
```

Comment:
Maybe just define as a const[] outside of the LoadingButtons declaration.

Then there is no need to memoize (which comes with its own overhead).